lib/utils/compose_ts.ts

```diff
@@ -730,6 +798,46 @@ export async function tarDirectory(
 	}
 }
 
+function setConfig(
+	configPath: string,
+	configData: MultiBuild.ParsedBalenaYml,
+	buildArgs: ParsedBuildArguments,
+): string {
+	const config = configData as any;
+	config['build-variables'] ||= {};
+
+	// Set global build variables if specified
+	if (!_.isEmpty(buildArgs.global)) {
+		config['build-variables'].global ||= [];
+		config['build-variables'].global = [
+			...config['build-variables'].global,
+			..._.map(buildArgs.global, (value: string, key: string) => {
+				return `${key}=${value}`;
+			}),
+		];
```

Comment:
Am I right that the use of array spread syntax here would allow multiple variables of the same name to remain in the array? For example, I reckon that the returned `config` object might end up with:
```
build-variables:
    global:
        - MY_VAR="Value defined in an existing '.balena/balena.yml' file"
        - MY_VAR="Overriding value provided with --buildArgs"
```
So two values for a single variable. In which case, the ultimate behavior would depend on how the cloud builders or balenaEngine or Docker v19 or Docker v20 or Docker Desktop vs Linux Docker choose to interpret it. This statement may be a bit exaggerated, but is there some truth in it? I am thinking that we are better off "resolving" the overriding behavior in the CLI, in this function, rather than passing the buck to builders / Docker. Unless you are really confident of Docker's / builder behavior, and could perhaps add a link to Docker API documentation that discusses it, which could then be added to the source code as a comment.

Otherwise, here's a suggestion: at least for the purpose of this overriding (but possibly also all the way through to stringification / serialization, given that `resin-multibuild` seems to be prepared to cope with it), instead of using `global: string[]`, we could adopt the `VarList` type of `resin-multibuild`, which I think is simply `{ [key: string]: string; }`, converting from `string[]` to `VarList` as needed, then use object spread syntax instead of array spread syntax:
```
global = { ...{ }, ...{ } }
```
instead of:
```
global = [ ...[ ], ...[ ] ]
```
because object spread syntax would take care of variable overriding for us in a way that array spread syntax doesn't:
```
global = {
    ...{ MY_VAR: "Value defined in an existing '.balena/balena.yml' file" },
    ...{ MY_VAR: "Overriding value provided with --buildArgs" },
}
```

Reply:
This is correct, updating the code use object spread syntax for overwrite.